File: employee-service/src/main/java/pl/piomin/services/employee/repository/EmployeeRepository.java
package pl.piomin.services.employee.repository;

import jakarta.inject.Singleton;
import pl.piomin.services.employee.model.Employee;

import java.util.ArrayList;
import java.util.List;
import java.util.Optional;
import java.util.stream.Collectors;

@Singleton
public class EmployeeRepository {

	private List<Employee> employees = new ArrayList<>();
	
	public Employee add(Employee employee) {
		employee.setId((long) (employees.size()+1));
		employees.add(employee);
		return employee;
	}
	
	public Employee findById(Long id) {
		Optional<Employee> employee = employees.stream().filter(a -> a.getId().equals(id)).findFirst();
		return employee.orElse(null);
	}
	
	public List<Employee> findAll() {
		return employees;
	}
	
	public List<Employee> findByDepartment(Long departmentId) {
		return employees.stream().filter(a -> a.getDepartmentId().equals(departmentId)).toList();
	}
	
	public List<Employee> findByOrganization(Long organizationId) {
		return employees.stream().filter(a -> a.getOrganizationId().equals(organizationId)).toList();
	}
	
}

File: employee-service/src/main/kotlin/pl/piomin/services/employee/repository/EmployeeRepository.kt
package pl.piomin.services.employee.repository

import io.micronaut.configuration.hibernate.jpa.scope.CurrentSession
import io.micronaut.spring.tx.annotation.Transactional
import pl.piomin.services.employee.model.Employee
import javax.inject.Singleton
import javax.persistence.EntityManager
import javax.persistence.PersistenceContext

@Singleton
open class EmployeeRepository(@param:CurrentSession @field:PersistenceContext val entityManager: EntityManager) {

    @Transactional
    open fun add(employee: Employee): Employee {
        entityManager.persist(employee)
        return employee
    }

    @Transactional(readOnly = true)
    open fun findById(id: Long): Employee = entityManager.find(Employee::class.java, id)

    @Transactional(readOnly = true)
    open fun findAll(): List<Employee> = entityManager.createQuery("SELECT e FROM Employee e").resultList as List<Employee>

    @Transactional(readOnly = true)
    open fun findByDepartment(departmentId: Long): List<Employee> = entityManager.createQuery("SELECT e FROM Employee e WHERE e.departmentId = :depId")
            .setParameter("depId", departmentId)
            .resultList as List<Employee>

    @Transactional(readOnly = true)
    open fun findByOrganization(organizationId: Long): List<Employee> = entityManager.createQuery("SELECT e FROM Employee e WHERE e.organizationId = :orgId")
            .setParameter("orgId", organizationId)
            .resultList as List<Employee>

}
